packages/controllers/src/utils/SIWXUtil.ts

```diff
@@ -197,16 +197,25 @@ export const SIWXUtil = {
 
     return sessions
   },
-  async getSessions() {
+  async getSessions(args?: { address?: string; caipNetworkId?: CaipNetworkId }) {
     const siwx = OptionsController.state.siwx
-    const address = CoreHelperUtil.getPlainAddress(ChainController.getActiveCaipAddress())
-    const network = ChainController.getActiveCaipNetwork()
+    let address = args?.address
+    if (!address) {
+      const activeCaipAddress = ChainController.getActiveCaipAddress()
+      address = CoreHelperUtil.getPlainAddress(activeCaipAddress)
```

Comment:
If not reused, 
```suggestion
      address = CoreHelperUtil.getPlainAddress(ChainController.getActiveCaipAddress())
```

Reply:
guess this is personal pref but I hate inlining function calls as it makes debugging a bit harder plus code looks uglier imo